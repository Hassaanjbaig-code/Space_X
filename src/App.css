* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;

  --color-primary: #0290ff;
  --color-primary-dark: #0290ff;
  --color-text: #121212;
  --color-text-blue: #227fdc;
  --color-text-gray: #888;
}

.flex-center {
  display: flex;
  justify-content: center;
  align-items: center;
}

.row {
  display: flex;
}

.column {
  display: flex;
  flex-direction: column;
}

.no-style {
  list-style: none;
  text-decoration: none;
  background-color: transparent;
  border: none;
}

.seprator {
  height: 15px;
  border: 1px solid #121212;
}

.App {
  text-align: center;
}
