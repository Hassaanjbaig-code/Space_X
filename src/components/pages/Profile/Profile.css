.profile-rockets,
.profile-missions {
  width: 100%;
}

.list-reserved {
  width: 80%;
  border: 1px solid #d3d3d3;
  border-radius: 12px;
  margin-top: 24px;
  min-height: 100px;
}

.list-reserved li {
  width: 100%;
  padding: 12px;
  border-bottom: 1px solid #d3d3d3;
}

.list-reserved li:last-child {
  border: none;
}
