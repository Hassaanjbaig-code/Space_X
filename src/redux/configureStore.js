import { configureStore } from '@reduxjs/toolkit';
import rocketReducer from './rockets/rocketReducer';
import missionReducer from './missions/missionReducer';

export default configureStore({
  reducer: {
    rocketReducer,
    mission: missionReducer.reducer,
  },
});
