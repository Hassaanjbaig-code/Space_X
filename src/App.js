import './App.css';
import Navbar from './components/Navbar';
import { Routes, Route } from 'react-router-dom';
import ListMission from './components/pages/ListMission';
import ListRockect from './components/pages/ListRockect';
import Profile from './components/pages/Profile';

function App() {
  return (
    <div className="App">
      <Navbar />
      <Routes>
        <Route path="/" element={<ListRockect />} />
        <Route path="/Missions" element={<ListMission />} />
        <Route path="/My Profile" element={<Profile />} />
      </Routes>
    </div>
  );
}

export default App;