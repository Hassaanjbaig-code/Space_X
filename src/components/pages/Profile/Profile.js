import React from 'react';
import { useSelector } from 'react-redux';
import './Profile.css';

const Profile = () => {
  const listRocket = useSelector((state) => state.rocketReducer.listRocket);
  const listMission = useSelector((state) => state.mission.data);

  return (
    <div className="flex-center">
      <div className="flex-center column profile-missions">
        <h2>mission</h2>
        <ul className="no-style flex-center column list-reserved">
          {listMission.length > 0 ? listMission.filter((data) => data.reserved).map((mission) => (
            <li key={mission.id} id={mission.id}>
              <h4>{mission.name}</h4>
            </li>
          )) : 'Empty List'}
        </ul>
      </div>
      <div className="flex-center column profile-rockets">
        <h2>My Rockets</h2>
        <ul className="no-style flex-center column list-reserved">
          {(listRocket && listRocket.filter((rocket) => rocket.reserved).length !== 0)
            ? listRocket.filter((rocket) => rocket.reserved)
              .map((rocket) => (<li key={rocket.id}>{rocket.rocket_name}</li>))
            : 'Empty List'}
        </ul>
      </div>
    </div>
  );
};

export default Profile;
