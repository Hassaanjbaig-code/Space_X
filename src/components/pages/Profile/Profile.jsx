import React, { useState } from 'react';
import { useSelector } from 'react-redux';
import './Profile.css'

const Profile = () => {
  const listRocket = useSelector(state=> state.rocketReducer.listRocket)

  return (<div className='flex-center'>
    <div className="flex-center column profile-missions">
    <h2>mission</h2>
    <ul className='no-style flex-center column list-reserved'>
    </ul>
    </div>
    <div className="flex-center column profile-rockets">
      <h2>My Rockets</h2>
      <ul className='no-style flex-center column list-reserved'>
        {(listRocket && listRocket.filter(rocket=> rocket.reserved).length != 0) 
        ? listRocket.filter(rocket=> rocket.reserved)
        .map(rocket => (<li key={rocket.id} >{rocket.rocket_name}</li>)) 
        : "Empty List"
        }
      </ul>
    </div>
  </div>);
};

export default Profile;
