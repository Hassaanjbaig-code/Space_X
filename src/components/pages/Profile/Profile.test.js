import renderer, { act } from 'react-test-renderer';
import '@testing-library/jest-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from '../../../redux/configureStore';
import Profile from './Profile';
import App from '../../../App';

const listRocketTest = [
  {
    id: '5e9d0d95eda69955f709d1eb',
    rocket_name: 'Falcon 1',
    description: 'The Falcon 1 was an expendable launch system priva…launch vehicle to go into orbit around the Earth.',
    flickr_images: 'https://imgur.com/DaCfMsj.jpg',
  },
  {
    id: '5e9d0d95eda69973a809d1ec',
    rocket_name: 'Falcon 9',
    description: 'Falcon 9 is a two-stage rocket designed and manufa… satellites and the Dragon spacecraft into orbit.',
    flickr_images: 'https://farm1.staticflickr.com/929/28787338307_3453a11a77_b.jpg',
  },
  {
    id: '5e9d0d95eda69974db09d1ed',
    rocket_name: 'Falcon Heavy',
    description: 'With the ability to lift into orbit over 54 metric…hicle, the Delta IV Heavy, at one-third the cost.',
    flickr_images: 'https://farm5.staticflickr.com/4599/38583829295_581f34dd84_b.jpg',
  },
  {
    id: '5e9d0d96eda699382d09d1ee',
    rocket_name: 'Starship',
    description: 'Starship and Super Heavy Rocket represent a fully …tually replace Falcon 9, Falcon Heavy and Dragon.',
    flickr_images: 'https://live.staticflickr.com/65535/48954138962_ee541e6755_b.jpg',
  },
];

const store = configureStore;

describe('test render of profile', () => {
  test('Profile should match snapshoot', () => {
    const tree = renderer.create(
      <React.StrictMode>
        <Provider store={store}>
          <Profile />
        </Provider>
      </React.StrictMode>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('test user interaction', () => {
  test('test reserve rocket', async () => {
    render(
      <React.StrictMode>
        <Provider store={store}>
          <Router>
            <App />
          </Router>
        </Provider>
      </React.StrictMode>,
    );

    // simulate a API REPOND to the reducer
    await act(async () => {
      store.dispatch({ type: 'spacex/rocket/GET_ROCKETS/fulfilled', payload: listRocketTest });
    });

    // test the first rocket
    const buttons = screen.getAllByRole('button');

    // reserve 3 rocket
    fireEvent.click(buttons[0]);
    fireEvent.click(buttons[1]);
    fireEvent.click(buttons[2]);

    // Navigate to profile page
    const profileLink = screen.getAllByRole('link')[2];
    fireEvent.click(profileLink);

    // check if all elements are displayed in the list of rocket
    expect(screen.getByText(listRocketTest[0].rocket_name)).toBeTruthy();
    expect(screen.getByText(listRocketTest[1].rocket_name)).toBeTruthy();
    expect(screen.getByText(listRocketTest[2].rocket_name)).toBeTruthy();
  });
});
