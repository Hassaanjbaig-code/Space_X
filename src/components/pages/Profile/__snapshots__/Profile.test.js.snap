// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test render of profile Profile should match snapshoot 1`] = `
<div
  className="flex-center"
>
  <div
    className="flex-center column profile-missions"
  >
    <h2>
      mission
    </h2>
    <ul
      className="no-style flex-center column list-reserved"
    >
      Empty List
    </ul>
  </div>
  <div
    className="flex-center column profile-rockets"
  >
    <h2>
      My Rockets
    </h2>
    <ul
      className="no-style flex-center column list-reserved"
    >
      Empty List
    </ul>
  </div>
</div>
`;
